lib/engine/game/g_18_ms.rb

```diff
@@ -62,12 +62,22 @@ def new_operating_round(round_num = 1)
           @phase.next! if @turn == 1 && round_num == 2
         end
 
+        if round_num == 1
+          @players.each do |p|
+            next unless p.cash.negative?
+
+            debt = -p.cash
+            interest = (debt / 2 + 0.99).floor
+            p.spend(interest, @bank, check_cash: false)
+            @log << "#{p.name} has to borrow another #{format_currency(interest)} as being in debt at end of SR"
+          end
+        end
         super
       end
 
       def operating_round(round_num)
         Round::Operating.new(self, [
-          Step::Bankrupt,
+          Step::G18MS::Bankrupt,
```

Comment:
why do we need step::bankrupt at all?can't you just not have anything

Reply:
When doing emergency buy, before you have sold all shares, you can press the "Declare Bankrupt" button in the view.

Removing the file would mean that you get some game exception as bankrupt action is not handled.

So to avoid this problem I had two choices:
- Remove the button (based on a [class] constant) - e.g show only if DECLARE_BANKRUPCY is true [by default this is true, 18MS it is false]
- If button pressed make a game_error raise explaining that you cannot declare bankrupcy but have to buy the train

With the option to remove the step from 18MS, which one do you prefer?

Reply:
can't you just check the available actions?

Reply:
Ah, yes it is a better solution. Do know not render the "Declare Bankruptcy" button if there is not step that handle process_bankrupt action.